actions/emoji_actions.js

```diff
@@ -0,0 +1,167 @@
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+
+import {setRecentEmojis} from 'actions/local_storage';
+import {getEmojiMap, getRecentEmojis, isCustomEmojiEnabled} from 'selectors/emojis';
+import {isCustomStatusEnabled, makeGetCustomStatus} from 'selectors/views/custom_status';
+
+import {ActionTypes} from 'utils/constants';
+import {EmojiIndicesByAlias} from 'utils/emoji';
+
+export function loadRecentlyUsedCustomEmojis() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const config = getConfig(state);
+
+        if (config.EnableCustomEmoji !== 'true') {
+            return {data: true};
+        }
+
+        const recentEmojis = getRecentEmojis(state);
+        const emojiMap = getEmojiMap(state);
+        const missingEmojis = recentEmojis.filter((name) => !emojiMap.has(name));
+
+        missingEmojis.forEach((name) => {
+            dispatch(EmojiActions.getCustomEmojiByName(name));
+        });
+
+        return {data: true};
+    };
+}
+
+export function incrementEmojiPickerPage() {
+    return async (dispatch) => {
+        dispatch({
+            type: ActionTypes.INCREMENT_EMOJI_PICKER_PAGE,
+        });
+
+        return {data: true};
+    };
+}
+
+const MAXIMUM_RECENT_EMOJI = 27;
+
+export function addRecentEmoji(alias) {
+    return (dispatch, getState) => {
+        const state = getState();
+        const recentEmojis = getRecentEmojis(state);
+        const emojiMap = getEmojiMap(state);
+
+        let name;
+        const emoji = emojiMap.get(alias);
+        if (!emoji) {
+            return;
+        } else if (emoji.name) {
+            name = emoji.name;
+        } else {
+            name = emoji.aliases[0];
+        }
+
+        const index = recentEmojis.indexOf(name);
+        if (index !== -1) {
+            recentEmojis.splice(index, 1);
+        }
+
+        recentEmojis.push(name);
+
+        if (recentEmojis.length > MAXIMUM_RECENT_EMOJI) {
+            recentEmojis.splice(0, recentEmojis.length - MAXIMUM_RECENT_EMOJI);
+        }
+
+        dispatch(setRecentEmojis(recentEmojis));
+    };
+}
+
+export function loadCustomEmojisForCustomStatusesByUserIds(userIds) {
+    return (dispatch, getState) => {
+        const state = getState();
+        const customEmojiEnabled = isCustomEmojiEnabled(state);
+        const customStatusEnabled = isCustomStatusEnabled(state);
+        if (!customEmojiEnabled || !customStatusEnabled) {
+            return {data: false};
+        }
+
+        const getCustomStatus = makeGetCustomStatus();
+        const emojisToLoad = new Set();
+
+        userIds.forEach((userId) => {
+            const customStatus = getCustomStatus(state, userId);
+            if (!customStatus || !customStatus.emoji) {
+                return;
+            }
+
+            emojisToLoad.add(customStatus.emoji);
+        });
+
+        dispatch(loadCustomEmojisIfNeeded(Array.from(emojisToLoad)));
```

Comment:
Minor nit, but we could just return the result of this dispatch to satisfy the typing and allow the inner action to return an error if necessary instead of returning `{data: true}` on the line below this.